inc/ThirdParty/Plugins/Optimization/AMP.php

```diff
@@ -0,0 +1,112 @@
+<?php
+namespace WP_Rocket\ThirdParty\Plugins\Optimization;
+
+use WP_Rocket\Admin\Options_Data;
+use WP_Rocket\Event_Management\Subscriber_Interface;
+
+/**
+ * Subscriber for compatibility with AMP
+ *
+ * @since  3.5.2
+ * @author Soponar Cristina
+ */
+class AMP implements Subscriber_Interface {
+	const QUERY       = 'amp';
+	const AMP_OPTIONS = 'amp-options';
+
+	/**
+	 * WP Rocket Options instance
+	 *
+	 * @var Options_Data
+	 */
+	private $options;
+
+	/**
+	 * Constructor
+	 *
+	 * @param Options_Data $options WP Rocket Options instance.
+	 */
+	public function __construct( Options_Data $options ) {
+		$this->options = $options;
+	}
+
+	/**
+	 * Subscribed events for AMP.
+	 *
+	 * @since  3.5.2
+	 * @author Soponar Cristina
+	 * @inheritDoc
+	 */
+	public static function get_subscribed_events() {
+		$events = [
+			'activate_amp/amp.php'   => 'generate_config_file',
+			'deactivate_amp/amp.php' => 'generate_config_file',
+		];
+		if ( function_exists( 'is_amp_endpoint' ) ) {
+			$events['wp']                         = 'disable_options_on_amp';
+			$events['rocket_cache_query_strings'] = 'is_amp_compatible_callback';
+			$events['update_option_amp-options']  = 'generate_config_file';
+		}
+
+		return $events;
+	}
+
+	/**
+	 * Regenerate config file on plugin activation / deactivation.
+	 *
+	 * @since  3.5.2
+	 * @author Soponar Cristina
+	 */
+	public function generate_config_file() {
+		rocket_generate_config_file();
+	}
+
+	/**
+	 * Add compatibility with AMP query string by adding it as a cached query string.
+	 *
+	 * @since  3.5.2
+	 * @author Soponar Cristina
+	 *
+	 * @param array $value WP Rocket cache_query_strings value.
+	 */
```

Comment:
Add `@return`

Reply:
done in https://github.com/wp-media/wp-rocket/pull/2459/commits/e69dcc1dcbe18a6976e86fa40b35c8175b8c956c